packages/chai-openapi-response-validator/test/assertions/satisfyApiSpec/specsDefineServersDifferently.test.js

```diff
@@ -493,4 +493,127 @@ describe('Using OpenAPI 3 specs that define servers differently', () => {
       });
     });
   });
+
+  describe('spec uses server variables to define server', () => {
+    before(() => {
+      const pathToApiSpec = path.join(
+        dirContainingApiSpec,
+        'withServerVariables.yml',
+      );
+      chai.use(chaiResponseValidator(pathToApiSpec));
+    });
+
+    describe("res.req.path matches the server path with a single variable", () => {
+      const res = {
+        status: 200,
+        req: {
+          method: 'GET',
+          path: 'https://test.com:1234/test/responseBody/string',
+        },
+        body: 'valid body (string)',
+      };
+
+      it('passes', () => {
+        expect(res).to.satisfyApiSpec;
+      });
+
+      it('fails when using .not', () => {
+        const assertion = () => expect(res).to.not.satisfyApiSpec;
+        expect(assertion).to.throw(AssertionError, '');
+      });
+    });
+
+    describe("res.req.path matches the server path with default variables and an endpoint path", () => {
+      const res = {
+        status: 200,
+        req: {
+          method: 'GET',
+          path: 'https://default.test.com:1234/test/responseBody/string',
+        },
+        body: 'valid body (string)',
+      };
+
+      it('passes', () => {
+        expect(res).to.satisfyApiSpec;
+      });
+
+      it('fails when using .not', () => {
+        const assertion = () => expect(res).to.not.satisfyApiSpec;
+        expect(assertion).to.throw(AssertionError, '');
+      });
+    });
+
+    describe("res.req.path matches the server path with non-default variables and an endpoint path", () => {
+      const res = {
+        status: 200,
+        req: {
+          method: 'GET',
+          path: 'https://nondefault.test.com:5678/test/responseBody/string',
+        },
+        body: 'valid body (string)',
+      };
+
+      it('passes', () => {
+        expect(res).to.satisfyApiSpec;
+      });
+
+      it('fails when using .not', () => {
+        const assertion = () => expect(res).to.not.satisfyApiSpec;
+        expect(assertion).to.throw(AssertionError, '');
+      });
+    });
+
+    describe("res.req.path matches the server path with default variables but no endpoint paths", () => {
+      const res = {
+        status: 200,
+        req: {
+          method: 'GET',
+          path: '/nonExistentEndpointPath',
+        },
+        body: 'valid body (string)',
+      };
+
+      it('fails', () => {
+        const assertion = () => expect(res).to.satisfyApiSpec;
+        expect(assertion).to.throw(
+          AssertionError,
+          `'/nonExistentEndpointPath' matches servers ${inspect([
+            'https://test.com:1234',
```

Comment:
I think we might actually want to be a little more specific than this, and it goes hand in hand with the comment you made elsewhere about only changing the server variables in the path.

I.e. in my tests I have a server with a port variable, and a path variable: `https://test.com:{portVariable}/{pathVariable}`. When I make a request with the path `/defaultPathVariable` at the moment I would get back:
```
      '/defaultPathVariable' matches servers [
        'https://test.com:1234/defaultPathVariable',
        'https://test.com:5678/defaultPathVariable'
      ]
```
Obviously the multiple servers with different port numbers is excessive, however if we changed it to 
```
      '/defaultPathVariable' matches servers [
        'https://test.com:{portVariable}/{pathVariable}'
      ]
```
Then it's actually too vague, since in reality the request only matches one of the `pathVariable` enums: `defaultPathVariable`, not all `pathVariable` enums.

Instead, if we do the work to only change the server variables in the path, then it should be possible to give the following error message, which I think is the preferred solution:

 ```
      '/defaultPathVariable' matches servers [
        'https://test.com:{portVariable}/defaultPathVariable'
      ]
```

Reply:
well put, I agree!